Sources/StreamChatUI/ChatMessageList/ChatMessage/Decorators/ChatMessageHeaderDecoratorView.swift

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+public struct ChatMessageHeaderDecoratorViewContent {
+    public let message: ChatMessage
+    public let channel: ChatChannel
+    public let dateFormatter: MessageDateSeparatorFormatter
+    public let shouldShowDate: Bool
+    public let shouldShowUnreadMessages: Bool
+
+    public init(
+        message: ChatMessage,
+        channel: ChatChannel,
+        dateFormatter: MessageDateSeparatorFormatter,
+        shouldShowDate: Bool,
+        shouldShowUnreadMessages: Bool
+    ) {
+        self.message = message
+        self.channel = channel
+        self.dateFormatter = dateFormatter
+        self.shouldShowDate = shouldShowDate
+        self.shouldShowUnreadMessages = shouldShowUnreadMessages
+    }
+}
+
+/// The decorator view that is used as a container for the chat message header view decorators.
+public final class ChatMessageHeaderDecoratorView: ChatMessageDecorationView, ThemeProvider {
```

Comment:
```suggestion
public final class ChatChannelMessageHeaderDecoratorView: ChatMessageDecorationView, ThemeProvider {
```
Same as above, as this is used by channels shall we rename it to contain the `Channel` in its name?